src/nvim/spell.c

```diff
@@ -752,9 +734,8 @@ static void find_word(matchinf_T *mip, int mode)
           // COMPOUNDRULE, discard the compounded word.
           continue;
         }
-      }
-      // Check NEEDCOMPOUND: can't use word without compounding.
-      else if (flags & WF_NEEDCOMP) {
+      } else if (flags & WF_NEEDCOMP) {
+        // Check NEEDCOMPOUND: can't use word without compounding.
```

Comment:
This looks easier to understand for me
```suggestion
        // Check NEEDCOMPOUND: can't use word without compounding.
      } else if (flags & WF_NEEDCOMP) {
```

Reply:
In some places moving the comment into the if statement doesn't look confusing, but it does look confusing here, as this sentence is written in an imperative tone.

Reply:
Then it's better to rewrite the comment, e.g.,
```
// skip if word is only valid in a compound
```